src/metabase/api/table.clj

```diff
@@ -64,11 +65,11 @@
         (t2/hydrate :db :pk_field)
         fix-schema)))
 
-(defn- update-table!*
+(mu/defn ^:private update-table!*
   "Takes an existing table and the changes, updates in the database and optionally calls `table/update-field-positions!`
   if field positions have changed."
-  [{:keys [id] :as existing-table} body]
-  {id ms/PositiveInt}
+  [{:keys [id] :as existing-table} :- [:map [:id ::lib.schema.id/table]]
+   body]
```

Comment:
Everything else in this PR is a fix for a Kondo warning from the latest version -- `{id ms/PositiveInt}` was an unused value here.